src/repos/MagazineRepo.ts

```diff
@@ -0,0 +1,177 @@
+) {
+  return MagazineModel.find({ semester })
+    .sort({ date: 'desc' })
+    .skip(offset)
+    .limit(limit)
+    .then((magazines) => {
+      return magazines.filter((magazine) => magazine !== null && !isMagazineFiltered(magazine));
+    });
+}
+
+function getMagazinesByPublicationSlug(
+  slug: string,
+  limit: number = DEFAULT_LIMIT,
+  offset: number = DEFAULT_OFFSET,
+) {
+  return MagazineModel.find({ publicationSlug: slug })
+    .sort({ date: 'desc' })
+    .skip(offset)
+    .limit(limit)
+    .then((magazines) => {
+      return magazines.filter((magazine) => magazine !== null && !isMagazineFiltered(magazine));
+    });
+}
+
+const getMagazinesByPublicationSlugs = async (
+  slugs: string[],
+  limit: number = DEFAULT_LIMIT,
+  offset: number = DEFAULT_OFFSET,
+): Promise<Magazine[]> => {
+  const uniqueSlugs = [...new Set(slugs)];
+  return MagazineModel.find({ 'publication.slug': { $in: uniqueSlugs } })
+    .sort({ date: 'desc' })
+    .skip(offset)
+    .limit(limit)
+    .then((magazines) => {
+      return magazines.filter((magazine) => magazine !== null && !isMagazineFiltered(magazine));
+    });
+};
+
+function getMagazineByID(id: string) {
+  return MagazineModel.findById(id).then((magazine) => {
+    if (!isMagazineFiltered(magazine)) {
+      return magazine;
+    }
+    return null;
+  });
+}
+
+function getMagazinesByIDs(ids: string[]) {
+  return Promise.all(ids.map((id) => MagazineModel.findById(new ObjectId(id)))).then(
+    (magazines) => {
+      return magazines.filter((magazine) => magazine !== null && !isMagazineFiltered(magazine));
+    },
+  );
+}
+
+/**
+ * Computes and returns the featured magazines in the database.
+ *
+ * @function
+ * @param {number} limit - number of magazines to retrieve.
+ */
+const getFeaturedMagazines = async (limit = DEFAULT_LIMIT): Promise<Magazine[]> => {
+  const magazines = await MagazineModel.find({ isFeatured: true }).exec();
+  return magazines.filter((magazine) => !isMagazineFiltered(magazine)).slice(0, limit);
+};
+
+/**
+ * Refreshes featured magazines.
+ */
+const refreshFeaturedMagazines = async (): Promise<Magazine[]> => {
+  // Set previous featured magazines to not featured
+  const oldFeaturedMagazines = await MagazineModel.find({ isFeatured: true }).exec();
+  oldFeaturedMagazines.forEach(async (m) => {
+    const magazine = await MagazineModel.findById(new ObjectId(m._id)); // eslint-disable-line
+    magazine.isFeatured = false;
+    await magazine.save();
+  });
+
+  // Get new featured magazines
+  const magazines = await MagazineModel.aggregate()
+    // Get a sample of random magazines
+    .sample(100)
+    // Get magazines after 30 days ago
+    .match({
+      date: {
+        $gte: new Date(
+          new Date().setDate(new Date().getDate() - MAX_NUM_DAYS_OF_FEATURED_MAGAZINES),
+        ),
+      },
+    });
+
+  magazines.forEach(async (m) => {
+    const magazine = await MagazineModel.findById(new ObjectId(m._id)); // eslint-disable-line
```

Comment:
why did we disable the linter for this line? if possible, we should disable a specific lint so it's clearer why it was disabled in the first place

Reply:
The linter had a rule against dangling underscores, added a file-wide disable for  no-underscore-dangle